packages/core/src/internal/services/i18n.service.ts

```diff
@@ -183,17 +184,42 @@ export class I18nService {
   static get keys(): Readonly<I18nStrings> {
     // intialize registry if not yet set
     if (Object.keys(GlobalStateService.state.i18nRegistry).length === 0) {
-      GlobalStateService.state.i18nRegistry = { ...componentStringsDefault, ...GlobalStateService.state.i18nRegistry };
+      GlobalStateService.state.i18nRegistry = mergeObjects(
+        componentStringsDefault,
+        GlobalStateService.state.i18nRegistry
+      );
+    }
+
+    return mergeObjects(componentStringsDefault, GlobalStateService.state.i18nRegistry) as I18nStrings;
+  }
+
+  static findKey(keyValueObj: any) {
+    const keyVals = I18nService.keys;
+    return Object.keys(keyVals).find(key => objectNaiveDeepEquals((keyVals as any)[key], keyValueObj));
```

Comment:
Previously we had a `===` here. That was failing. This was the root cause of the overrides overwriting the i18n keys. Fixed now.